crypto/evp/keymgmt_lib.c

```diff
@@ -156,7 +157,8 @@ void *evp_keymgmt_util_export_to_provider(EVP_PKEY *pk, EVP_KEYMGMT *keymgmt)
         return NULL;
     }
 
-    CRYPTO_THREAD_write_lock(pk->lock);
+    if (!CRYPTO_THREAD_write_lock(pk->lock))
```

Comment:
Should evp_keymgmt_freedata() get called?

Reply:
> Should evp_keymgmt_freedata() get called?

good catch.  updated fixup commit pushed.